lib/accesscontrol/permitted_action_on_object_query.rb

```diff
@@ -31,6 +31,30 @@ def can?(action_id, object_type, object_id)
       end
     end
 
+    # Allow permission on an ActiveRecord object.
+    # A grant is universally unique and is enforced at the database level.
+    #
+    # @param action_id [Integer] The action to grant for the object
+    # @param object_type [ActiveRecord::Base] The ActiveRecord model that receives a permission grant.
+    # @param object_id [Integer] The id of the ActiveRecord object which receives a permission grant
+    # @return [nil] Returns nil if successful, otherwise will raise an Error (AccessControl::CouldNotGrantError).
+    #
+    # @example
+    #   # Allow the user access to Post 7
+    #   AccessControl::Query.new(user).grant(3, Post, 7)
+    def grant(action_id, object_type, object_id)
+      a = PermittedActionOnObject.create(
+        id: SecureRandom.uuid,
+        actor: @actor,
+        action: action_id,
+        object_type: String(object_type),
+        object_id: object_id
+      )
+      nil
+    rescue
+      raise AccessControl::CouldNotGrantError.new("Could not grant action #{action_id} on object #{object_type} with id #{object_id} for actor #{@actor}")
```

Comment:
Do you think we should silently `rescue` from `PG::UniqueViolation` (or the equivalent on other DBMSs if we end up supporting them)? Technically, if that error is raised, then the permission has already been granted and nothing *really* wrong has happened, right?

Of course, if other errors happened, then the caller definitely needs to know about it. In those cases, should we bubble up or include the error message so it's clear what the problem is?

Reply:
Should we just return `nil` on the PG::UniqueViolation?

Also, if we start using `Postgres` specific errors we will need to update our test suite to run against a `Postgres` database, right?

Reply:
> Should we just return nil on the PG::UniqueViolation?

Makes sense to me!

> Also, if we start using Postgres specific errors we will need to update our test suite to run against a Postgres database, right?

Bah, you're right! Is there a more generic way we can do it?

Reply:
I don't know there's an easier way - but since we're leveraging `where_tuples` we may just want to support Postgres to start with?

Reply:
Oh sweet!